code/mob/suicide.dm

```diff
@@ -21,7 +21,7 @@
 // the suiciding var is already at the mob level for fuck's sakes
 /mob/verb/suicide()
 
-	if (!isliving(src) || isdead(src))
+	if ((!isliving(src) || isdead(src)) && !istype(src, /mob/dead/aieye))
```

Comment:
Do you need this if you've got your override in place?

Reply:
Unfortunately yea, the aieye mob passes the isdead check because it checks for stat == 2 (aieye has stat = 0), but the isliving check will only pass if the mob is a subtype of /mob/living which aieye isn't. The alternative was changing isliving to isalive which checks for stat == 0 but I didn't want to inadvertently create new bugs in case something in the code doesn't set stat properly.

Reply:
Ah, yep, I see it was the `do_suicide` proc that was overridden, rather than the `suicide` verb.